src/core/include/openvino/core/op_extension.hpp

```diff
@@ -100,4 +100,6 @@ class OpExtension : public BaseOpExtension {
     }
 };
 
+/** @brief Class to distinguish legacy extension. */
+class LegacyOpExtension : public BaseOpExtension {};
```

Comment:
Can we move this class to private or developer API?

Reply:
I think, yes. 
Should be done by some macro or moved to `detail` namespace?

Reply:
I think we can just move it to the same place which is used for SOExtension for example.

Reply:
Done.